### DISCUSSION TITLE
Profile T fails at start of device conformance test

### DISCUSSION THREAD

**Discussion by parsley72** [onvif/testspecs#61] [Q&A] [answered]
When I run the ONVIF Device Test Tool (22.12 rev 28) Conformance Test it does a basic check then prints:
```
Feature definition process PASSED (27 supported, 255 unsupported, 0 undefined)
Profile Support Preliminary Check:
   Profile S SUPPORTED
   Profile A NOT SUPPORTED
   Profile D NOT SUPPORTED
   Profile M NOT SUPPORTED
   Profile C NOT SUPPORTED
   Profile G NOT SUPPORTED
   Profile T FAILED
Not all profiles claimed are supported, conformance will be FAILED
```
and proceeds to run all the relevant tests. But why is Profile T failing before it runs the tests? Is there any logging I can see that would explain?

**Comment by madhurao68** [comment]
Hello @parsley72

The test result clearly shows that 255 features are unsupported which indicates that several mandatory Profile T features may have failed. You can refer the Profile feature overview available at https://www.onvif.org/wp-content/uploads/2022/04/onvif-profile-feature-overview.pdf?441d4a&441d4a to check the mandatory features for Profile T.
 
If conformance test is throwing errors, click on '**Generate All Reports with Errata**' to see a list of errors with their numbers and descriptions.

![image](https://github.com/onvif/testspecs/assets/73811069/b3b3cd9d-e104-4fd0-b3fd-b9cd469e376b)

The tool will then generate and save Declaration of Conformance, Test Report, and Feature List with the following name formats:

- ONVIF_DoC_produtname_firmware_timestamp.pdf
- ONVIF_TestReport_Product name_firmware_timestamp>.pdf
- ONVIF_FeatureList_Product name_firmware_timestamp>.xml


![image](https://github.com/onvif/testspecs/assets/73811069/58b2f5e8-6360-4843-b84e-d096f3a66044)

You can check the errors to find the cause of the failure.  Please refer the Help file (How do I) available in the Device test tool for a more detailed explanation.

Hope this helps!

**Comment by parsley72** [comment]
I can easily see the tests that failed in the Diagnostics tab. What I'm asking is why Profile T is shown as failing **before the tests are run**. The test app is obviously querying the device to see what functionality is supported and checking this against a list for the profiles. How can I see this debug?

**Reply by baldvin137** [reply]
The Device Test Tool does a feature discovery and then compares the features discovered to the Profile specifications.

[ONVIF Device Feature Discovery Specification](https://www.onvif.org/wp-content/uploads/2022/07/ONVIF_Device_Feature_Discovery_Specification_21.12.pdf) 

[ONVIF Profile Conformance Device Test Specification](https://www.onvif.org/wp-content/uploads/2022/07/ONVIF_Profiles_Conformance_Device_Test_Specification_21.12.pdf)

**Comment by madhurao68** [comment, marked as answer]
The process followed for conformance is as follows:

1. Check features during Feature discovery process - DTT checks which features are claimed as supported. The results are available at Diagnostic -> Features tab.
2. Check Profiles support - DTT checks which profiles are claimed as supported and if Device has all required features for this. The results are available at Diagnostic -> Profiles tab.
3. Check features implementation by executing Test Cases.  The results are available at Diagnostic -> Test Cases tab.

If one of the above fails conformance fails.
In this case something is wrong with step 2. You can check the log by opening Diagnostic -> Profiles tab and selecting the failed profile.
